sihl_core/lib/db.ml

```diff
@@ -127,44 +118,20 @@ let query_db_exn ?message request query =
   | Error msg -> Fail.raise_database (Option.value ~default:msg message)
 
 module Migrate = struct
-  type migration_error = Caqti_error.t
-
-  type migration_operation =
-    Caqti_lwt.connection -> unit -> (unit, migration_error) Result.t Lwt.t
-
-  type migration_step = string * migration_operation
-
-  type migration = string * migration_step list
+  include Db_migration_core
 
   module State = struct
-    module Model = struct
-      type t = { namespace : string; version : int; dirty : bool }
-      [@@deriving fields]
-
-      let create ~namespace = { namespace; version = 0; dirty = true }
-
-      let mark_dirty state = { state with dirty = true }
-
-      let mark_clean state = { state with dirty = false }
-
-      let increment state = { state with version = state.version + 1 }
-
-      let steps_to_apply (namespace, steps) { version; _ } =
-        (namespace, List.drop steps version)
-    end
-
-    module Repository = struct
+    module PostgresRepository : Contract.Migration.REPOSITORY = struct
```

Comment:
I'm not very fond of having DBMS specific code here, but we can think how to pull it out later.

Reply:
Agree, this is temporary.